### PR TITLE
feat(backup/gcs): store can be configured

### LINKED ISSUE #11924: GCS backup store can be configured
Rough idea for required configuration:
```yaml
zeebe.broker.data.backup:
  store: gcs
  gcs:
    bucketName:
    basePath:
```

At least two items will be added later:
- Authentication
- Compression

### PR DESCRIPTION
**Pull request by lenaschoenburg**
Adds initial configuration for the GCS backup store:

```yaml
zeebe.broker.data.backup:
  store: gcs
  gcs:
    bucketName:
    basePath:
```

Broker and Restore app use the GCS store when `zeebe.broker.data.backup.store: gcs`.
Initial implementation of GCS store is empty, all methods throw `UnsupportedOperationException`.

closes #11924

### CODE REVIEW

File: dist/src/main/config/broker.standalone.yaml.template [hunk 1]
```diff
@@ -332,6 +333,19 @@
...
 
+        # Configure the following if store is set to GCS
+        # gcs:
+          # Name of the bucket where the backup will be stored.
+          # The bucket must already exist.
+          # The bucket must not be shared with other Zeebe clusters unless basePath is also set.
+          # This setting can also be overridden using the environment variable ZEEBE_BROKER_DATA_BACKUP_GCS_BUCKETNAME
+          # bucketName:
+
+          # When set, all blobs in the bucket will use this prefix. Must be non-empty and not start or end with '/'.
```
Comment (Zelldon) [inline dist/src/main/config/broker.standalone.yaml.template:344]:
```suggestion
          # When set, all blobs in the bucket will use this prefix. Must be non-empty and not start or end with '/'.
```

For better UX we could also just remove the trailing `/` right?

**Review by ChrisKujawa — APPROVED** [review]
thanks looks good :) :+1:

File: dist/src/main/config/broker.standalone.yaml.template (hunk 1 above)
Comment (oleschoenburg) [reply dist/src/main/config/broker.standalone.yaml.template]: :+1: I implemented and tested this in the last two commits

**Pull request merged**